aws/provider.go

```diff
@@ -293,7 +293,7 @@ func Provider() terraform.ResourceProvider {
 			"aws_cognito_identity_pool_roles_attachment":   resourceAwsCognitoIdentityPoolRolesAttachment(),
 			"aws_cognito_user_pool":                        resourceAwsCognitoUserPool(),
 			"aws_cognito_user_pool_domain":                 resourceAwsCognitoUserPoolDomain(),
-			"aws_autoscaling_lifecycle_hook":               resourceAwsAutoscalingLifecycleHook(),
+			"aws_cognito_user_pool_client":                 resourceAwsCognitoUserPoolClient(),
```

Comment:
I believe that the removal of `aws_autoscaling_lifecycle_hook` was not intended here - I'll address that in a separate commit.